include/revng/TupleTree/Tracking.h

```diff
@@ -183,6 +233,11 @@ struct Tracking {
   static void pop(const M &LHS) {
     visitTuple<M, PopVisitor>(LHS);
   }
+
+  template<typename M>
+  static void stop(const M &LHS) {
+    visitTuple<M, StopTrackingVisitor>(LHS);
```

Comment:
i not sure i understand the rationale behind this one. how does it differ from  my version?

Reply:
You don't need a class, you can just use a lambda.

Reply:
i don't understand how this is supposed to work.
did you meant
```
visit([]<typename T>(const T &Obj){
   if contexper (IsSetOrKOC<T>) 
     Obj.stopTracking() 
});
```

Reply:
If that's necessary, yup. But in these cases I'd just use `auto` instead of `<typename T>`.
But yeah, whatever works with lambda is good.

Reply:
as we said this one it is possible to write the visitors with a inline lambda, but since we have a custom visitor here doing so would require to write a lambda that is longer and uglier than writing a the struct like they are now. in the end i would keep this one like this.

Reply:
previous message was incorrect. this is not writable with a lamda because one of the visitor methods has two template parameters and the other has 1.